lib/core/webdriver/firefox.js

```diff
@@ -242,6 +236,18 @@ module.exports.configureBuilder = function(builder, options) {
 
   profile.setPreference('devtools.chrome.enabled', true);
 
+  if (options.firefox && options.firefox.preference) {
+    const userPrefs = util.toArray(options.firefox.preference);
+    for (const pref of userPrefs) {
+        const nameAndValue = pref.split(':');
+        if (nameAndValue.length === 2) {
+          profile.setPreference(nameAndValue[0], nameAndValue[1]);
+        } else {
+          log.error('Firefox preferences %s need to of the format key:value, preference was not set', pref)
+        }
+    }
+  }
+
   const firefoxConfig = options.firefox || {};
```

Comment:
If you move this just above the if statement, there's no need for ```if (options.firefox```. Also util.toArray handles undefined, to you can skip the if altogether and just have ```const userPrefs = util.toArray(firefoxConfig.preference);```